core/src/test/java/org/apache/calcite/test/enumerable/EnumerableCorrelateTest.java

```diff
@@ -168,6 +168,24 @@ class EnumerableCorrelateTest {
             "empid=200");
   }
 
+  /** Test case for
+   * <a href="https://issues.apache.org/jira/browse/CALCITE-5638">[CALCITE-5638]
+   * Complex nested correlated subquery failed</a>.
+   */
+  @Test void complexNestedCorrelatedSubquery() {
+    String sql = "SELECT empid, deptno, (SELECT count(*) FROM emps AS x "
+        + "WHERE x.salary>emps.salary and x.deptno<emps.deptno) FROM emps "
+        + "WHERE empid<salary ORDER BY 1,2,3";
+
+    tester(false, new HrSchema())
```

Comment:
I assume this has been validated using another database?

Reply:
of course )